---
title: "RTCGA package tutorial"
author: "Marcin Kosinski"
date: "May 5, 2015"
output:
  html_document:
    theme: united
    highlight: tango
    fig_width: 17
    fig_height: 10
    toc: true
    toc_depth: 4
    keep_md: true
---


```{r, echo=FALSE}
library(knitr)
opts_chunk$set(comment="", message=FALSE, warning = FALSE, tidy.opts=list(keep.blank.line=TRUE, width.cutoff=150),options(width=150), cache=TRUE)
```


# Mission

The Cancer Genome Atlas (TCGA) Data Portal provides a platform for researchers to search, download, and analyze data sets generated by TCGA. It contains clinical information, genomic characterization data, and high level sequence analysis of the tumor genomes[1]. The key is to understand genomics to improve cancer care. 

The RTCGA package offers an easy interface for downloading and integrating variety of the TCGA data using patient barcode key. This allows for easier data acquisition facilitating development of science and improvement of patients' treatment. RTCGA is an open-source R package available to download from Bioconductor[2]. Furthermore, the RTCGA package transforms the TCGA data to a form which is (most ?) convenient to use with R statistical packages like (give an example). The data transformation can be a part of the statistical analysis pipeline which can be more reproducible with RTCGA (nie rozumiem tego zdania).


# RTCGA package

More detail information about this package can be found
here [https://github.com/MarcinKosinski/RTCGA](https://github.com/MarcinKosinski/RTCGA).


## Installation of the RTCGA package
To get started, install the latest version of **RTCGA** from Bioconductor:
```{r }
# not there yet
```
or use:
```{r, eval=FALSE}
if (!require(devtools)) {
    install.packages("devtools")
    library(devtools)
}
install_github("MarcinKosinski/RTCGA")
```

Make sure you have [rtools](http://cran.r-project.org/bin/windows/Rtools/) installed on your computer.

# Data manipulations

## Available cohorts

```{r}
library(RTCGA)
(cohorts <- infoTCGA() %>% 
   names() %>% 
   sub("-counts", "", x=.))
```



## Downloading data

One can check what is the newest available date of release of TCGA datasets.
```{r}

tail( availableDates(), 2 )
date <- tail( availableDates() ,1 )
```

For that date we can download datasets for **ALL** Cohorts to the
`data` folder.

### Clinical data

Can be downloaded using default settings. By default the dataset is downloaded from the newest available date of TCGA releases.
```{r}
#dir.create( "data" )
downloadTCGA( cancerTypes = cohorts, destDir = "data/", date = date )
```


Let us check the names and sizes of the downloaded files.

```{r}
file.info( paste0( "data/", grep( x = list.files("data/"), pattern = "gdac", value = TRUE )) )[, 1:2]
```


### Mutations data

Let us check if there is Mutations data available for all canser types.

```{r, cache = FALSE, message=FALSE, warning=FALSE}
library(RTCGA)
library(dplyr)
checkDataSetsAvailability( cohorts, 
                           pattern = "Mutation_Packager_Calls.Level", 
                           date = date) %>%
   unlist() %>% 
   cat( sep="\n")

```


Let us download it.
```{r}
sapply( cohorts, function(element){
tryCatch({
downloadTCGA( cancerTypes = element, 
              dataSet = "Mutation_Packager_Calls.Level",
              destDir = "data/", 
              date = date )},
error = function(cond){
   cat("Error: Maybe there weren't mutations data for ", element, " cancer.\n")
}
)
})
```


Let us check the names and sizes of downloaded the files.


```{r}
file.info( paste0( "data/", grep( x = list.files("data/"), pattern = "gdac", value = TRUE )) )[, 1:2]
```

### RNA-Seq data

Let us check if there is RNA-Seq data available for all canser types.

```{r }

checkDataSetsAvailability( cohorts,
                        "rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_genes_normalized__data.Level",
                        date = date ) %>%
   unlist() %>%  
   cat( sep="\n")

```


Let us download it.

```{r}
sapply( cohorts, function(element){
tryCatch({
downloadTCGA( cancerTypes = element, 
              dataSet = "rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_genes_normalized__data.Level",
              destDir = "data/", 
              date = date )},
error = function(cond){
   cat("Error: Maybe there weren't rnaseq data for ", element, " cancer.\n")
}
)
}) 
```

Let us check the names and sizes of downloaded files.


```{r}
file.info( paste0( "data/", grep( x = list.files("data/"), pattern = "gdac", value = TRUE )) )[, 1:2] 
```

## Untarring data

Let us use the `untar()` function to untar all downloaded sets.

```{r, warning=FALSE, results='hide'}

list.files( "data/") %>% 
   paste0( "data/", .) %>%
   sapply( untar, exdir = "data/" )

```


### Removing no longer needed `tar.gz` files

After datasets are untarred, the `tar.gz` files ar no longer needed and can be deleted.

```{r, results='hide'}
list.files( "data/") %>% 
   paste0( "data/", .) %>%
   grep( pattern = "tar.gz", x = ., value = TRUE) %>%
   sapply( file.remove )
```



## Combining data

### Available genes in RNA-Seq data


Let us check the available genes names for **ALL** cancers
in `rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_genes_normalized__data.Level` datasets




```{r, cache=FALSE, echo=3:19, eval=FALSE}
# addd interactive DT from DT/rstudio package

genes <- list.files( "data/") %>%
   grep( pattern = "rnaseqv2__illuminahiseq", x = ., value = TRUE ) %>%
   paste0( "data/", . , "/") %>%
   sapply( function( element ){ 
      paste0( element, list.files( element ) ) }) %>%
   sapply( availableGenesNames ) %>%
   .[which( (lapply(., length) > 0) == TRUE ) ] %>%
   as.data.frame( ) 

names(genes) <- cohorts
library(DT)

datatable( genes , options=list(iDisplayLength = 10) )


```

### Merging Clinical Data and RNA-Seq data

Now, let us add information about **MDM2** expression into the `.clinical.merged.txt` files,
for each cancer type. Note that you can pass an abbreviation of the name of a gene as a parameter and then all matching genes will be added (example: for **TP53** 7 genes are added) at the end of `.clinical.merged.txt` file.

#### Paths to clinical data

```{r}

```

```{r, width.cutoff=120}

sapply( cohorts, function( element ){
   folder <- grep( paste0( "(_",element,"\\.", "|","_",element,"-FFPE)", ".*Clinical"), list.files("data/"),value = TRUE)
   path <- paste0( "data/", folder, "/", element, ".clin.merged.txt")
   assign( value = path, x = paste0(element, ".clin.path"), envir = .GlobalEnv)
}) 

```


#### Paths to rnaseq data

```{r}
sapply( cohorts, function( element ){
   folder <- grep( paste0( "(_",element,"\\.", "|","_",element,"-FFPE)", ".*rnaseqv2"), list.files("data/"),value = TRUE)
   file <- grep( paste0(element, ".*rnaseqv2"), list.files( paste0( "data/",folder ) ),
                 value = TRUE)
   path <- paste0( "data/", folder, "/", file )
   assign( value = path, x = paste0(element, ".rnaseq.path"), envir = .GlobalEnv)
}) 
```


```{r, warning=FALSE}


sapply( cohorts, function( element ){
   tryCatch({
   mergeTCGA(
      clinicalDir = get(paste0(element,".clin.path"), envir = .GlobalEnv),
      rnaseqDir = get(paste0(element,".rnaseq.path"), envir = .GlobalEnv),
      genes = c("MDM2")
   )},
   error=function(cond){
   cat("Error: Maybe there weren't rnaseq data for ", element, " cancer.\n")
})
})


```



### Merging Clinical Data and Mutations data

#### Paths to Muations data


```{r}
sapply( cohorts, function( element ){
   folder <- grep( paste0( "(_",element,"\\.", "|","_",element,"-FFPE)", ".*Mutation_Packager"), list.files("data/"),value = TRUE)
   path <- paste0( "data/", folder, "/")
   assign( value = path, x = paste0(element, ".mutations.dir"), envir = .GlobalEnv)
})
```

Let us add information about the **TP53** gene mutations to `*clinical.merged.txt` files.

```{r}
sapply( cohorts, function( element ){
   tryCatch({
   mergeTCGA(
      clinicalDir = get(paste0(element,".clin.path"), envir = .GlobalEnv),
      mutationDir = get(paste0(element,".mutations.dir"), envir = .GlobalEnv),
      genes = c("TP53")
   )},
   error=function(cond){
   cat("Error: Maybe there weren't mutations data for ", element, " cancer.\n")
})
})

```



## Joining all clinical.merged.files into one data set


```{r}
sapply( cohorts, function(element){
   tryCatch({
    assign( value = read.delim(get(paste0(element,".clin.path"), envir = .GlobalEnv)),
    x = paste0(element, ".clin.data"),
    , envir = .GlobalEnv)
    dim(get(x = paste0(element, ".clin.data"),
    , envir = .GlobalEnv))
   },
   error=function(cond){
   cat("Error: Maybe there weren't clinical data for ", element, " cancer.\n")
})
})
```

```{r}

# ClinicalDataNames <- grep( "clin.data", x = ls(), value = TRUE )
# ClinicalDataNames
# clinicalCombo <- get(ClinicalDataNames[1], envir = .GlobalEnv)
# cat(ClinicalDataNames, "\n")
# cat(dim(clinicalCombo), "\n")
# for( i in seq_along(ClinicalDataNames)[-1]){
#    clinicalCombo <- full_join( clinicalCombo, 
#                                get(ClinicalDataNames[i], envir = .GlobalEnv), 
#                                by = "V1")
#    cat(ClinicalDataNames[i], "\n")
#    cat(dim(clinicalCombo), "\n") #different names? but not all :) !)
# }
# if one wants to perform for loop or sapply then one
# NEEDS TO REMEMBER TO CHAGE NAMES EVERy 4th full_join.....

names(ACC.clin.data)[-1] <- paste0("ACC.clin.data", 2:ncol(ACC.clin.data))
names(BLCA.clin.data)[-1] <- paste0("BLCA.clin.data", 2:ncol(ACC.clin.data))
clinicalCombo <- full_join( ACC.clin.data, BLCA.clin.data, by = "V1")
names(clinicalCombo)[-1] <- paste0("clinicalCombo", 2:ncol(clinicalCombo))
clinicalCombo <- full_join( clinicalCombo, BRCA.clin.data, by = "V1")
names(clinicalCombo)[-1] <- paste0("clinicalCombo", 2:ncol(clinicalCombo))
clinicalCombo <- full_join( clinicalCombo, CESC.clin.data, by = "V1")
names(clinicalCombo)[-1] <- paste0("clinicalCombo", 2:ncol(clinicalCombo))
clinicalCombo <- full_join( clinicalCombo, CHOL.clin.data, by = "V1")
names(clinicalCombo)[-1] <- paste0("clinicalCombo", 2:ncol(clinicalCombo))
clinicalCombo <- full_join( clinicalCombo, COAD.clin.data, by = "V1")
names(clinicalCombo)[-1] <- paste0("clinicalCombo", 2:ncol(clinicalCombo))
#clinicalCombo <- full_join( clinicalCombo, COADREAD.clin.data, by = "V1") COAD + READ
clinicalCombo <- full_join( clinicalCombo, DLBC.clin.data, by = "V1")
names(clinicalCombo)[-1] <- paste0("clinicalCombo", 2:ncol(clinicalCombo))
clinicalCombo <- full_join( clinicalCombo, ESCA.clin.data, by = "V1")
names(clinicalCombo)[-1] <- paste0("clinicalCombo", 2:ncol(clinicalCombo))
#clinicalCombo <- full_join( clinicalCombo, FPPP.clin.data, by = "V1") # many
clinicalCombo <- full_join( clinicalCombo, GBM.clin.data, by = "V1")
names(clinicalCombo)[-1] <- paste0("clinicalCombo", 2:ncol(clinicalCombo))
#clinicalCombo <- full_join( clinicalCombo, GBMLGG.clin.data, by = "V1") # gbm + lgg
clinicalCombo <- full_join( clinicalCombo, HNSC.clin.data, by = "V1")
names(clinicalCombo)[-1] <- paste0("clinicalCombo", 2:ncol(clinicalCombo))
clinicalCombo <- full_join( clinicalCombo, KICH.clin.data, by = "V1")
names(clinicalCombo)[-1] <- paste0("clinicalCombo", 2:ncol(clinicalCombo))
#clinicalCombo <- full_join( clinicalCombo, KIPAN.clin.data, by = "V1") # kich kirc kirp
clinicalCombo <- full_join( clinicalCombo, KIRC.clin.data, by = "V1")
names(clinicalCombo)[-1] <- paste0("clinicalCombo", 2:ncol(clinicalCombo))
clinicalCombo <- full_join( clinicalCombo, KIRP.clin.data, by = "V1")
names(clinicalCombo)[-1] <- paste0("clinicalCombo", 2:ncol(clinicalCombo))
clinicalCombo <- full_join( clinicalCombo, LAML.clin.data, by = "V1")
names(clinicalCombo)[-1] <- paste0("clinicalCombo", 2:ncol(clinicalCombo))
clinicalCombo <- full_join( clinicalCombo, LGG.clin.data, by = "V1")
names(clinicalCombo)[-1] <- paste0("clinicalCombo", 2:ncol(clinicalCombo))
clinicalCombo <- full_join( clinicalCombo, LIHC.clin.data, by = "V1")
names(clinicalCombo)[-1] <- paste0("clinicalCombo", 2:ncol(clinicalCombo))
clinicalCombo <- full_join( clinicalCombo, LUAD.clin.data, by = "V1")
names(clinicalCombo)[-1] <- paste0("clinicalCombo", 2:ncol(clinicalCombo))
clinicalCombo <- full_join( clinicalCombo, LUSC.clin.data, by = "V1")
names(clinicalCombo)[-1] <- paste0("clinicalCombo", 2:ncol(clinicalCombo))
clinicalCombo <- full_join( clinicalCombo, MESO.clin.data, by = "V1")
names(clinicalCombo)[-1] <- paste0("clinicalCombo", 2:ncol(clinicalCombo))
clinicalCombo <- full_join( clinicalCombo, OV.clin.data, by = "V1")
names(clinicalCombo)[-1] <- paste0("clinicalCombo", 2:ncol(clinicalCombo))
clinicalCombo <- full_join( clinicalCombo, PAAD.clin.data, by = "V1")
names(clinicalCombo)[-1] <- paste0("clinicalCombo", 2:ncol(clinicalCombo))
clinicalCombo <- full_join( clinicalCombo, PCPG.clin.data, by = "V1")
names(clinicalCombo)[-1] <- paste0("clinicalCombo", 2:ncol(clinicalCombo))
clinicalCombo <- full_join( clinicalCombo, READ.clin.data, by = "V1")
names(clinicalCombo)[-1] <- paste0("clinicalCombo", 2:ncol(clinicalCombo))
clinicalCombo <- full_join( clinicalCombo, SKCM.clin.data, by = "V1")
names(clinicalCombo)[-1] <- paste0("clinicalCombo", 2:ncol(clinicalCombo))
clinicalCombo <- full_join( clinicalCombo, STAD.clin.data, by = "V1")
names(clinicalCombo)[-1] <- paste0("clinicalCombo", 2:ncol(clinicalCombo))
clinicalCombo <- full_join( clinicalCombo, TGCT.clin.data, by = "V1")
names(clinicalCombo)[-1] <- paste0("clinicalCombo", 2:ncol(clinicalCombo))
clinicalCombo <- full_join( clinicalCombo, THCA.clin.data, by = "V1")
names(clinicalCombo)[-1] <- paste0("clinicalCombo", 2:ncol(clinicalCombo))
clinicalCombo <- full_join( clinicalCombo, THYM.clin.data, by = "V1")
names(clinicalCombo)[-1] <- paste0("clinicalCombo", 2:ncol(clinicalCombo))
clinicalCombo <- full_join( clinicalCombo, UCEC.clin.data, by = "V1")
names(clinicalCombo)[-1] <- paste0("clinicalCombo", 2:ncol(clinicalCombo))
clinicalCombo <- full_join( clinicalCombo, UCS.clin.data, by = "V1")
names(clinicalCombo)[-1] <- paste0("clinicalCombo", 2:ncol(clinicalCombo))
clinicalCombo <- full_join( clinicalCombo, UVM.clin.data, by = "V1")
names(clinicalCombo)[-1] <- paste0("clinicalCombo", 2:ncol(clinicalCombo))

table(unlist(as.matrix(clinicalCombo[which(clinicalCombo$V1 == "admin.disease_code"),])))

```



Let us save it into new file 


```{r}
file.create("clinicalComboHuge.txt")
write.table(clinicalCombo,
            file = "clinicalComboHuge.txt",
            quote = FALSE,
            sep = "\t",
            row.names = FALSE)
```

Reimport
```{r}
clinicalCombo <- read.clinical("clinicalCombo.txt")
```





