# Mission

The Cancer Genome Atlas (TCGA) Data Portal provides a platform for researchers to search, download, and analyze data sets generated by TCGA. It contains clinical information, genomic characterization data, and high level sequence analysis of the tumor genomes[1]. The key is to understand genomics to improve cancer care. 

RTCGA package offers download and integration of the variety and volume of TCGA data using patient barcode key, what enables easier data possession. This may have an benefcial infuence on impact on development of science and improvement of patients' treatment. RTCGA is an open-source R package, available to download from Bioconductor[2]. Furthermore, RTCGA package transforms TCGA data to form which is convenient to use in R statistical package. Those data transformations can be a part of statistical analysis pipeline which can be more reproducible with RTCGA.

# RTCGA

>
> Clinical and rna-seq data downloaded with the usage of `RTCGA` are available in [`RTCGA.data`](https://github.com/mi2-warsaw/RTCGA.data) package.
>

Project is (will be) supported by [Travis CI](https://travis-ci.org/) and [waffle.io](https://waffle.io/).

<h5> Installation of the RTCGA package: </h5>
To get started, install the latest version of **RTCGA** from Bioconductor:
```{Ruby}
# not there yet
```
or use:
```{Ruby}
if (!require(devtools)) {
    install.packages("devtools")
    require(devtools)
}
install_github("MarcinKosinski/RTCGA")
```
Make sure you have [rtools](http://cran.r-project.org/bin/windows/Rtools/) installed on your computer.

<h5> The list of available functions: </h5>
```{Ruby}
help(package="RTCGA")
```

**To run examples below use those packages**
```{Ruby}
library(RTCGA)
library(dplyr)
```

<h5> Information about available TCGA data:</h5>
```{Ruby}
infoTCGA()
```

<h5> Cohorts names:</h5>
```{Ruby}
(cohorts <- infoTCGA() %>% 
   names() %>% 
   sub("-counts", "", x=.))
```

Cohorts codes are explained here: [http://gdac.broadinstitute.org/](http://gdac.broadinstitute.org/).

<h5> Dates of reales: </h5>
```{Ruby}
availableDates()
```
It is always save to use almost the last date, because sometimes data from the newest date are not available in 100%. So that's why I suggest using such date:
```{Ruby}
date <- tail( availableDates(), 2 )[1]
```

<h3> Clinical data</h3>
Can be downloaded using default settings. By default the dataset is downloaded from the newest available date of TCGA releases.

```{Ruby}
#dir.create( "data" )
downloadTCGA( cancerTypes = cohorts, destDir = "data/", date = date )
```
<h3> Mutations data</h3>
Let us check if there are Mutations data available for all canser types.

```{Ruby}
checkDataSetsAvailability( cohorts, 
                           pattern = "Mutation_Packager_Calls.Level", 
                           date = date) %>%
   unlist() %>% 
   cat( sep="\n")
```
Let us download them.
```{Ruby}
sapply( cohorts, function(element){
   tryCatch({
      downloadTCGA( cancerTypes = element, 
                    dataSet = "Mutation_Packager_Calls.Level",
                    destDir = "data/", 
                    date = date )},
      error = function(cond){
         cat("Error: Maybe there weren't mutations data for ", element, " cancer.\n")
   })
})
```
<h3> Genes expressions - RNA-Seq data </h3>
Let us check if there are RNA-Seq data available for all canser types.
```{Ruby}
checkDataSetsAvailability( cohorts,
                        "rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_genes_normalized__data.Level",
                        date = date ) %>%
   unlist() %>%  
   cat( sep="\n")
```

Let us download them.
```{Ruby}
sapply( cohorts, function(element){
   tryCatch({
      downloadTCGA( cancerTypes = element, 
                    dataSet = "rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_genes_normalized__data.Level",
                    destDir = "data/", 
                    date = date )},
      error = function(cond){
         cat("Error: Maybe there weren't rnaseq data for ", element, " cancer.\n")
   })
}) 
```


<h3> Untarring data </h3>

Letâ€™s use `untar()` function to untar all downloaded sets.
```{Ruby}
list.files( "data/") %>% 
   paste0( "data/", .) %>%
   sapply( untar, exdir = "data/" )
```

<h3> Removing no longer needed `tar.gz` files </h3>

After datasets are untarred, the `tar.gz` files ar no longer needed and can be deleted.
```{Ruby}
list.files( "data/") %>% 
   paste0( "data/", .) %>%
   grep( pattern = "tar.gz", x = ., value = TRUE) %>%
   sapply( file.remove )
```


<h4> The list of use-cases: </h4>

TODO.


