<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Information About Datasets from TCGA Project — checkTCGA • RTCGA</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">

<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">


<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script>
<script src="../pkgdown.js"></script>

<!-- mathjax -->
<script src='https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">RTCGA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../articles/index.html">Articles</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/RTCGA/RTCGA">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/kosinski_rblog">
    <span class="fa fa-twitter"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

      <div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Information About Datasets from TCGA Project</h1>
    </div>

    
    <p>The <code>checkTCGA</code> function let&#39;s to check</p>

    <p><ul>
<li> <code>DataSets</code>: TCGA datasets&#39; names for current release date and cohort.
     </li>
<li> <code>Dates</code>: TCGA datasets&#39; dates of release.
 </li>
</ul></p>
    

    <pre><span class='fu'>checkTCGA</span>(<span class='no'>what</span>, <span class='no'>cancerType</span>, <span class='kw'>date</span> <span class='kw'>=</span> <span class='kw'>NULL</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a> Arguments</h2>
    <dl class="dl-horizontal">
      <dt>what</dt>
      <dd>One of <code>DataSets</code> or <code>Dates</code>.</dd>
      <dt>cancerType</dt>
      <dd>A character of length 1 containing abbreviation (Cohort code - <a href = 'http://gdac.broadinstitute.org/'>http://gdac.broadinstitute.org/</a>)
of types of cancers to check for.</dd>
      <dt>date</dt>
      <dd>A <code>NULL</code> or character specifying from which date informations should be checked.
By default (<code>date = NULL</code>) the newest available date is used. All available dates can be checked on 
<a href = 'http://gdac.broadinstitute.org/runs/'>http://gdac.broadinstitute.org/runs/</a> or by using <code>checkTCGA(&#39;Dates&#39;)</code> 
function. Required format <code>&#39;YYYY-MM-DD&#39;</code>.</dd>
    </dl>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p><ul>
<li> If <code>what=&#39;DataSets&#39;</code> a data.frame of available datasets&#39; names (to pass to the <a href='downloadTCGA.html'>downloadTCGA</a> function) and sizes.
     </li>
<li> If <code>what=&#39;Dates&#39;</code> a vector of available dates to pass to the <a href='downloadTCGA.html'>downloadTCGA</a> function.
     </li>
</ul></p>
    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p><ul>
<li> If <code>what=&#39;DataSets&#39;</code> enables to check TCGA datasets&#39; names for current release date and cohort.
     </li>
<li> If <code>what=&#39;Dates&#39;</code> enables to check dates of TCGA datasets&#39; releases.
     </li>
</ul></p>
    
    <h2 class="hasAnchor" id="issues"><a class="anchor" href="#issues"></a>Issues</h2>

    <p></p>
    <p>If you have any problems, issues or think that something is missing or is not
clear please post an issue on 
<a href = 'https://github.com/RTCGA/RTCGA/issues'>https://github.com/RTCGA/RTCGA/issues</a>.</p>
    
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <p><span class="pkg">RTCGA</span> website <a href = 'http://rtcga.github.io/RTCGA/'>http://rtcga.github.io/RTCGA/</a>.</p>
    <p>Other RTCGA: <code><a href='RTCGA-package.html'>RTCGA-package</a></code>,
  <code><a href='boxplotTCGA.html'>boxplotTCGA</a></code>, <code><a href='convertTCGA.html'>convertTCGA</a></code>,
  <code><a href='createTCGA.html'>createTCGA</a></code>, <code><a href='datasetsTCGA.html'>datasetsTCGA</a></code>,
  <code><a href='downloadTCGA.html'>downloadTCGA</a></code>,
  <code><a href='expressionsTCGA.html'>expressionsTCGA</a></code>, <code><a href='heatmapTCGA.html'>heatmapTCGA</a></code>,
  <code><a href='infoTCGA.html'>infoTCGA</a></code>, <code><a href='installTCGA.html'>installTCGA</a></code>,
  <code><a href='kmTCGA.html'>kmTCGA</a></code>, <code><a href='mutationsTCGA.html'>mutationsTCGA</a></code>,
  <code><a href='pcaTCGA.html'>pcaTCGA</a></code>, <code><a href='readTCGA.html'>readTCGA</a></code>,
  <code><a href='survivalTCGA.html'>survivalTCGA</a></code>, <code><a href='theme_RTCGA.html'>theme_RTCGA</a></code></p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='co'>############################# </span>

<span class='co'># names for current release date and cohort</span>
<span class='fu'>checkTCGA</span>(<span class='st'>'DataSets'</span>, <span class='st'>'BRCA'</span>)</div><div class='output co'>#&gt;    Size
#&gt; 1   37M
#&gt; 2   50K
#&gt; 3  723K
#&gt; 4  135K
#&gt; 5  135K
#&gt; 6   77K
#&gt; 7   78K
#&gt; 8  1.5M
#&gt; 9   57K
#&gt; 10 1.2K
#&gt; 11 160K
#&gt; 12 3.4M
#&gt; 13  86K
#&gt; 14  83M
#&gt; 15 3.2G
#&gt; 16 1.1M
#&gt; 17  15M
#&gt; 18 2.9M
#&gt; 19  44M
#&gt; 20 1.3M
#&gt; 21 2.6G
#&gt; 22 277M
#&gt; 23 195M
#&gt; 24 298M
#&gt; 25  93M
#&gt; 26 869M
#&gt; 27 249M
#&gt; 28 2.8G
#&gt; 29 243M
#&gt; 30  18M
#&gt; 31  18M
#&gt; 32 5.3M
#&gt; 33 4.6M
#&gt; 34  37M
#&gt; 35 399M
#&gt; 36  10M
#&gt; 37 1.1G
#&gt; 38  81M
#&gt; 39 1.9M
#&gt; 40  37M
#&gt; 41 1.5G
#&gt; 42 6.6M
#&gt; 43 7.4M
#&gt;                                                                                                                                             Name
#&gt; 1  BRCA-FFPE.Merge_methylation__humanmethylation450__jhu_usc_edu__Level_3__within_bioassay_data_set_function__data.Level_3.2016012800.0.0.tar.gz
#&gt; 2                   BRCA-FFPE.Merge_mirnaseq__illuminahiseq_mirnaseq__bcgsc_ca__Level_3__miR_gene_expression__data.Level_3.2016012800.0.0.tar.gz
#&gt; 3                BRCA-FFPE.Merge_mirnaseq__illuminahiseq_mirnaseq__bcgsc_ca__Level_3__miR_isoform_expression__data.Level_3.2016012800.0.0.tar.gz
#&gt; 4                         BRCA-FFPE.Merge_snp__genome_wide_snp_6__broad_mit_edu__Level_3__segmented_scna_hg18__seg.Level_3.2016012800.0.0.tar.gz
#&gt; 5                         BRCA-FFPE.Merge_snp__genome_wide_snp_6__broad_mit_edu__Level_3__segmented_scna_hg19__seg.Level_3.2016012800.0.0.tar.gz
#&gt; 6      BRCA-FFPE.Merge_snp__genome_wide_snp_6__broad_mit_edu__Level_3__segmented_scna_minus_germline_cnv_hg18__seg.Level_3.2016012800.0.0.tar.gz
#&gt; 7      BRCA-FFPE.Merge_snp__genome_wide_snp_6__broad_mit_edu__Level_3__segmented_scna_minus_germline_cnv_hg19__seg.Level_3.2016012800.0.0.tar.gz
#&gt; 8                                                                                 BRCA-FFPE.Methylation_Preprocess.Level_3.2016012800.0.0.tar.gz
#&gt; 9                                                                               BRCA-FFPE.miRseq_Mature_Preprocess.Level_3.2016012800.0.0.tar.gz
#&gt; 10                                                                                     BRCA-FFPE.miRseq_Preprocess.Level_3.2016012800.0.0.tar.gz
#&gt; 11                                                                                        BRCA.Clinical_Pick_Tier1.Level_4.2016012800.0.0.tar.gz
#&gt; 12                                                                                             BRCA.Merge_Clinical.Level_1.2016012800.0.0.tar.gz
#&gt; 13                              BRCA.Merge_cna__illuminahiseq_dnaseqc__hms_harvard_edu__Level_3__segmentation__seg.Level_3.2016012800.0.0.tar.gz
#&gt; 14       BRCA.Merge_methylation__humanmethylation27__jhu_usc_edu__Level_3__within_bioassay_data_set_function__data.Level_3.2016012800.0.0.tar.gz
#&gt; 15      BRCA.Merge_methylation__humanmethylation450__jhu_usc_edu__Level_3__within_bioassay_data_set_function__data.Level_3.2016012800.0.0.tar.gz
#&gt; 16                          BRCA.Merge_mirnaseq__illuminaga_mirnaseq__bcgsc_ca__Level_3__miR_gene_expression__data.Level_3.2016012800.0.0.tar.gz
#&gt; 17                       BRCA.Merge_mirnaseq__illuminaga_mirnaseq__bcgsc_ca__Level_3__miR_isoform_expression__data.Level_3.2016012800.0.0.tar.gz
#&gt; 18                       BRCA.Merge_mirnaseq__illuminahiseq_mirnaseq__bcgsc_ca__Level_3__miR_gene_expression__data.Level_3.2016012800.0.0.tar.gz
#&gt; 19                    BRCA.Merge_mirnaseq__illuminahiseq_mirnaseq__bcgsc_ca__Level_3__miR_isoform_expression__data.Level_3.2016012800.0.0.tar.gz
#&gt; 20                     BRCA.Merge_protein_exp__mda_rppa_core__mdanderson_org__Level_3__protein_normalization__data.Level_3.2016012800.0.0.tar.gz
#&gt; 21                                BRCA.Merge_rnaseq__illuminahiseq_rnaseq__unc_edu__Level_3__exon_expression__data.Level_3.2016012800.0.0.tar.gz
#&gt; 22                                BRCA.Merge_rnaseq__illuminahiseq_rnaseq__unc_edu__Level_3__gene_expression__data.Level_3.2016012800.0.0.tar.gz
#&gt; 23                     BRCA.Merge_rnaseq__illuminahiseq_rnaseq__unc_edu__Level_3__splice_junction_expression__data.Level_3.2016012800.0.0.tar.gz
#&gt; 24                                 BRCA.Merge_rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_genes__data.Level_3.2016012800.0.0.tar.gz
#&gt; 25                      BRCA.Merge_rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_genes_normalized__data.Level_3.2016012800.0.0.tar.gz
#&gt; 26                              BRCA.Merge_rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_isoforms__data.Level_3.2016012800.0.0.tar.gz
#&gt; 27                   BRCA.Merge_rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_isoforms_normalized__data.Level_3.2016012800.0.0.tar.gz
#&gt; 28                        BRCA.Merge_rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__exon_quantification__data.Level_3.2016012800.0.0.tar.gz
#&gt; 29                    BRCA.Merge_rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__junction_quantification__data.Level_3.2016012800.0.0.tar.gz
#&gt; 30                             BRCA.Merge_snp__genome_wide_snp_6__broad_mit_edu__Level_3__segmented_scna_hg18__seg.Level_3.2016012800.0.0.tar.gz
#&gt; 31                             BRCA.Merge_snp__genome_wide_snp_6__broad_mit_edu__Level_3__segmented_scna_hg19__seg.Level_3.2016012800.0.0.tar.gz
#&gt; 32          BRCA.Merge_snp__genome_wide_snp_6__broad_mit_edu__Level_3__segmented_scna_minus_germline_cnv_hg18__seg.Level_3.2016012800.0.0.tar.gz
#&gt; 33          BRCA.Merge_snp__genome_wide_snp_6__broad_mit_edu__Level_3__segmented_scna_minus_germline_cnv_hg19__seg.Level_3.2016012800.0.0.tar.gz
#&gt; 34       BRCA.Merge_transcriptome__agilentg4502a_07_3__unc_edu__Level_3__unc_lowess_normalization_gene_level__data.Level_3.2016012800.0.0.tar.gz
#&gt; 35                                                                                     BRCA.Methylation_Preprocess.Level_3.2016012800.0.0.tar.gz
#&gt; 36                                                                                    BRCA.Mutation_Packager_Calls.Level_3.2016012800.0.0.tar.gz
#&gt; 37                                                                                 BRCA.Mutation_Packager_Coverage.Level_3.2016012800.0.0.tar.gz
#&gt; 38                                                                          BRCA.Mutation_Packager_Oncotated_Calls.Level_3.2016012800.0.0.tar.gz
#&gt; 39                                                                                      BRCA.RPPA_AnnotateWithGene.Level_3.2016012800.0.0.tar.gz
#&gt; 40                                                                                     BRCA.mRNA_Preprocess_Median.Level_3.2016012800.0.0.tar.gz
#&gt; 41                                                                                         BRCA.mRNAseq_Preprocess.Level_3.2016012800.0.0.tar.gz
#&gt; 42                                                                                   BRCA.miRseq_Mature_Preprocess.Level_3.2016012800.0.0.tar.gz
#&gt; 43                                                                                          BRCA.miRseq_Preprocess.Level_3.2016012800.0.0.tar.gz</div><div class='input'><span class='co'>## Not run: ------------------------------------</span>
<span class='co'># checkTCGA('DataSets', 'OV', tail(checkTCGA('Dates'))[3])</span>
<span class='co'># #checkTCGA('DataSets', 'OV', checkTCGA('Dates')[5]) # error</span>
<span class='co'>## ---------------------------------------------</span>
<span class='co'># dates of TCGA datasets' releases.</span>
<span class='fu'>checkTCGA</span>(<span class='st'>'Dates'</span>)</div><div class='output co'>#&gt;  [1] &quot;2011-10-26&quot; &quot;2011-11-15&quot; &quot;2011-11-28&quot; &quot;2011-12-06&quot; &quot;2011-12-30&quot;
#&gt;  [6] &quot;2012-01-10&quot; &quot;2012-01-24&quot; &quot;2012-02-17&quot; &quot;2012-03-06&quot; &quot;2012-03-21&quot;
#&gt; [11] &quot;2012-04-12&quot; &quot;2012-04-25&quot; &quot;2012-05-15&quot; &quot;2012-05-25&quot; &quot;2012-06-06&quot;
#&gt; [16] &quot;2012-06-23&quot; &quot;2012-07-07&quot; &quot;2012-07-25&quot; &quot;2012-08-04&quot; &quot;2012-08-25&quot;
#&gt; [21] &quot;2012-09-13&quot; &quot;2012-10-04&quot; &quot;2012-10-18&quot; &quot;2012-10-20&quot; &quot;2012-10-24&quot;
#&gt; [26] &quot;2012-11-02&quot; &quot;2012-11-14&quot; &quot;2012-12-06&quot; &quot;2012-12-21&quot; &quot;2013-01-16&quot;
#&gt; [31] &quot;2013-02-03&quot; &quot;2013-02-22&quot; &quot;2013-03-09&quot; &quot;2013-03-26&quot; &quot;2013-04-06&quot;
#&gt; [36] &quot;2013-04-21&quot; &quot;2013-05-08&quot; &quot;2013-05-23&quot; &quot;2013-06-06&quot; &quot;2013-06-23&quot;
#&gt; [41] &quot;2013-07-15&quot; &quot;2013-08-09&quot; &quot;2013-09-23&quot; &quot;2013-10-10&quot; &quot;2013-11-14&quot;
#&gt; [46] &quot;2013-12-10&quot; &quot;2014-01-15&quot; &quot;2014-02-15&quot; &quot;2014-03-16&quot; &quot;2014-04-16&quot;
#&gt; [51] &quot;2014-05-18&quot; &quot;2014-06-14&quot; &quot;2014-07-15&quot; &quot;2014-09-02&quot; &quot;2014-10-17&quot;
#&gt; [56] &quot;2014-12-06&quot; &quot;2015-02-02&quot; &quot;2015-02-04&quot; &quot;2015-04-02&quot; &quot;2015-06-01&quot;
#&gt; [61] &quot;2015-08-21&quot; &quot;2015-11-01&quot; &quot;2016-01-28&quot;</div><div class='input'>
<span class='co'>############################# </span>
<span class='co'>## Not run: ------------------------------------</span>
<span class='co'># # TCGA datasets' names availability for </span>
<span class='co'># # current release date and cancer type.</span>
<span class='co'># </span>
<span class='co'># releaseDate &lt;- '2015-08-21'</span>
<span class='co'># cancerTypes &lt;- c('OV', 'BRCA')</span>
<span class='co'># </span>
<span class='co'># cancerTypes %&gt;% sapply(function(element){</span>
<span class='co'>#   grep(x = checkTCGA('DataSets', element, releaseDate)[, 1], </span>
<span class='co'>#       pattern = 'humanmethylation450', value = TRUE) %&gt;%</span>
<span class='co'>#        as.vector()</span>
<span class='co'>#        })</span>
<span class='co'>#        </span>
<span class='co'>## ---------------------------------------------</span></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#details">Details</a></li>

      <li><a href="#issues">Issues</a></li>

      <li><a href="#see-also">See also</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    
Marcin Kosinski, <a href='mailto:m.p.kosinski@gmail.com'>m.p.kosinski@gmail.com</a>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Marcin Kosinski.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  </body>
</html>
