<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Gather Expressions for TCGA Datasets — expressionsTCGA • RTCGA</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">

<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">


<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script>
<script src="../pkgdown.js"></script>

<!-- mathjax -->
<script src='https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">RTCGA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../articles/index.html">Articles</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/RTCGA/RTCGA">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/kosinski_rblog">
    <span class="fa fa-twitter"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

      <div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Gather Expressions for TCGA Datasets</h1>
    </div>

    
    <p>Function gathers expressions over multiple TCGA datasets and extracts expressions for desired genes.
See <a href='http://www.rdocumentation.org/packages/RTCGA.rnaseq/topics/rnaseq'>rnaseq</a>, <a href='http://www.rdocumentation.org/packages/RTCGA.mRNA/topics/mRNA'>mRNA</a>, <a href='http://www.rdocumentation.org/packages/RTCGA.RPPA/topics/RPPA'>RPPA</a>, <a href='http://www.rdocumentation.org/packages/RTCGA.miRNASeq/topics/miRNASeq'>miRNASeq</a>, <a href='http://www.rdocumentation.org/packages/RTCGA.methylation/topics/methylation'>methylation</a>.</p>
    

    <pre><span class='fu'>expressionsTCGA</span>(<span class='no'>...</span>, <span class='kw'>extract.cols</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>extract.names</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a> Arguments</h2>
    <dl class="dl-horizontal">
      <dt>...</dt>
      <dd>A data.frame or data.frames from TCGA study containing expressions informations.</dd>
      <dt>extract.cols</dt>
      <dd>A character specifing the names of columns to be extracted with <code>bcr_patient_barcode</code>. 
If <code>NULL</code> (by default) all columns are returned.</dd>
      <dt>extract.names</dt>
      <dd>Logical, whether to extract names of passed data.frames in <code>...</code>.</dd>
    </dl>
    
    <h2 class="hasAnchor" id="note"><a class="anchor" href="#note"></a>Note</h2>

    <p>Input data.frames should contain column <code>bcr_patient_barcode</code> if <code>extract.cols</code> is specified.</p>
    
    <h2 class="hasAnchor" id="issues"><a class="anchor" href="#issues"></a>Issues</h2>

    <p></p>
    <p>If you have any problems, issues or think that something is missing or is not
clear please post an issue on 
<a href = 'https://github.com/RTCGA/RTCGA/issues'>https://github.com/RTCGA/RTCGA/issues</a>.</p>
    
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <p><span class="pkg">RTCGA</span> website <a href = 'http://rtcga.github.io/RTCGA/Visualizations.html'>http://rtcga.github.io/RTCGA/Visualizations.html</a>.</p>
    <p>Other RTCGA: <code><a href='RTCGA-package.html'>RTCGA-package</a></code>,
  <code><a href='boxplotTCGA.html'>boxplotTCGA</a></code>, <code><a href='checkTCGA.html'>checkTCGA</a></code>,
  <code><a href='convertTCGA.html'>convertTCGA</a></code>, <code><a href='createTCGA.html'>createTCGA</a></code>,
  <code><a href='datasetsTCGA.html'>datasetsTCGA</a></code>, <code><a href='downloadTCGA.html'>downloadTCGA</a></code>,
  <code><a href='heatmapTCGA.html'>heatmapTCGA</a></code>, <code><a href='infoTCGA.html'>infoTCGA</a></code>,
  <code><a href='installTCGA.html'>installTCGA</a></code>, <code><a href='kmTCGA.html'>kmTCGA</a></code>,
  <code><a href='mutationsTCGA.html'>mutationsTCGA</a></code>, <code><a href='pcaTCGA.html'>pcaTCGA</a></code>,
  <code><a href='readTCGA.html'>readTCGA</a></code>, <code><a href='survivalTCGA.html'>survivalTCGA</a></code>,
  <code><a href='theme_RTCGA.html'>theme_RTCGA</a></code></p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='co'>## for all examples</span>
<span class='fu'>library</span>(<span class='no'>dplyr</span>)
<span class='fu'>library</span>(<span class='no'>tidyr</span>)
<span class='fu'>library</span>(<span class='no'>ggplot2</span>)

<span class='co'>## RNASeq expressions</span>
<span class='fu'>library</span>(<span class='no'>RTCGA.rnaseq</span>)
<span class='fu'>expressionsTCGA</span>(<span class='no'>BRCA.rnaseq</span>, <span class='no'>OV.rnaseq</span>, <span class='no'>HNSC.rnaseq</span>,
               <span class='kw'>extract.cols</span> <span class='kw'>=</span> <span class='st'>"VENTX|27287"</span>) <span class='kw'>%&gt;%</span>
  <span class='fu'>rename</span>(<span class='kw'>cohort</span> <span class='kw'>=</span> <span class='no'>dataset</span>,
         <span class='kw'>VENTX</span> <span class='kw'>=</span> <span class='no'>`VENTX|27287`</span>) <span class='kw'>%&gt;%</span>
 <span class='fu'>filter</span>(<span class='fu'>substr</span>(<span class='no'>bcr_patient_barcode</span>, <span class='fl'>14</span>, <span class='fl'>15</span>) <span class='kw'>==</span> <span class='st'>"01"</span>) <span class='kw'>%&gt;%</span> <span class='co'>#cancer samples</span>
  <span class='fu'>ggplot</span>(<span class='fu'>aes</span>(<span class='kw'>y</span> <span class='kw'>=</span> <span class='fu'>log1p</span>(<span class='no'>VENTX</span>),
             <span class='kw'>x</span> <span class='kw'>=</span> <span class='fu'>reorder</span>(<span class='no'>cohort</span>, <span class='fu'>log1p</span>(<span class='no'>VENTX</span>), <span class='no'>median</span>),
             <span class='kw'>fill</span> <span class='kw'>=</span> <span class='no'>cohort</span>)) +
  <span class='fu'>geom_boxplot</span>() +
  <span class='fu'><a href='theme_RTCGA.html'>theme_RTCGA</a></span>() +
  <span class='fu'>scale_fill_brewer</span>(<span class='kw'>palette</span> <span class='kw'>=</span> <span class='st'>"Dark2"</span>)</div><div class='output co'>#&gt; <span class='message'>Scale for &#39;fill&#39; is already present. Adding another scale for &#39;fill&#39;, which</span>
#&gt; <span class='message'>will replace the existing scale.</span></div><img src='expressionsTCGA-3.png' alt='' width='540' height='400' /><div class='input'>
<span class='co'>## mRNA expressions  </span>
<span class='fu'>library</span>(<span class='no'>tidyr</span>)
<span class='fu'>library</span>(<span class='no'>RTCGA.mRNA</span>)
<span class='fu'>expressionsTCGA</span>(<span class='no'>BRCA.mRNA</span>, <span class='no'>COAD.mRNA</span>, <span class='no'>LUSC.mRNA</span>, <span class='no'>UCEC.mRNA</span>,
               <span class='kw'>extract.cols</span> <span class='kw'>=</span> <span class='fu'>c</span>(<span class='st'>"ARHGAP24"</span>, <span class='st'>"TRAV20"</span>)) <span class='kw'>%&gt;%</span>
  <span class='fu'>rename</span>(<span class='kw'>cohort</span> <span class='kw'>=</span> <span class='no'>dataset</span>) <span class='kw'>%&gt;%</span>
  <span class='fu'>select</span>(-<span class='no'>bcr_patient_barcode</span>) <span class='kw'>%&gt;%</span>
  <span class='fu'>gather</span>(<span class='kw'>key</span> <span class='kw'>=</span> <span class='st'>"mRNA"</span>, <span class='kw'>value</span> <span class='kw'>=</span> <span class='st'>"value"</span>, -<span class='no'>cohort</span>)  <span class='kw'>%&gt;%</span>
  <span class='fu'>ggplot</span>(<span class='fu'>aes</span>(<span class='kw'>y</span> <span class='kw'>=</span> <span class='no'>value</span>,
             <span class='kw'>x</span> <span class='kw'>=</span> <span class='fu'>reorder</span>(<span class='no'>cohort</span>, <span class='no'>value</span>, <span class='no'>mean</span>),
             <span class='kw'>fill</span> <span class='kw'>=</span> <span class='no'>cohort</span>)) +
  <span class='fu'>geom_boxplot</span>() +
  <span class='fu'><a href='theme_RTCGA.html'>theme_RTCGA</a></span>() +
  <span class='fu'>scale_fill_brewer</span>(<span class='kw'>palette</span> <span class='kw'>=</span> <span class='st'>"Set3"</span>) +
  <span class='fu'>facet_grid</span>(<span class='no'>mRNA</span>~<span class='no'>.</span>) +
  <span class='fu'>theme</span>(<span class='kw'>legend.position</span> <span class='kw'>=</span> <span class='st'>"top"</span>)</div><div class='output co'>#&gt; <span class='message'>Scale for &#39;fill&#39; is already present. Adding another scale for &#39;fill&#39;, which</span>
#&gt; <span class='message'>will replace the existing scale.</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Removed 2 rows containing non-finite values (stat_boxplot).</span></div><img src='expressionsTCGA-7.png' alt='' width='540' height='400' /><div class='input'>

<span class='co'>## RPPA expressions</span>
<span class='fu'>library</span>(<span class='no'>RTCGA.RPPA</span>)
<span class='fu'>expressionsTCGA</span>(<span class='no'>ACC.RPPA</span>, <span class='no'>BLCA.RPPA</span>, <span class='no'>BRCA.RPPA</span>,
    <span class='kw'>extract.cols</span> <span class='kw'>=</span> <span class='fu'>c</span>(<span class='st'>"4E-BP1_pS65"</span>, <span class='st'>"4E-BP1"</span>)) <span class='kw'>%&gt;%</span>
  <span class='fu'>rename</span>(<span class='kw'>cohort</span> <span class='kw'>=</span> <span class='no'>dataset</span>) <span class='kw'>%&gt;%</span>
  <span class='fu'>select</span>(-<span class='no'>bcr_patient_barcode</span>) <span class='kw'>%&gt;%</span>
  <span class='fu'>gather</span>(<span class='kw'>key</span> <span class='kw'>=</span> <span class='st'>"RPPA"</span>, <span class='kw'>value</span> <span class='kw'>=</span> <span class='st'>"value"</span>, -<span class='no'>cohort</span>)  <span class='kw'>%&gt;%</span>
  <span class='fu'>ggplot</span>(<span class='fu'>aes</span>(<span class='kw'>fill</span> <span class='kw'>=</span> <span class='no'>cohort</span>,
             <span class='kw'>y</span> <span class='kw'>=</span> <span class='no'>value</span>,
             <span class='kw'>x</span> <span class='kw'>=</span> <span class='no'>RPPA</span>)) +
  <span class='fu'>geom_boxplot</span>() +
  <span class='fu'>theme_dark</span>(<span class='kw'>base_size</span> <span class='kw'>=</span> <span class='fl'>15</span>) +
  <span class='fu'>scale_fill_manual</span>(<span class='kw'>values</span> <span class='kw'>=</span> <span class='fu'>c</span>(<span class='st'>"#eb6420"</span>, <span class='st'>"#207de5"</span>, <span class='st'>"#fbca04"</span>)) +
  <span class='fu'>coord_flip</span>() +
  <span class='fu'>theme</span>(<span class='kw'>legend.position</span> <span class='kw'>=</span> <span class='st'>"top"</span>) +
  <span class='fu'>geom_jitter</span>(<span class='kw'>alpha</span> <span class='kw'>=</span> <span class='fl'>0.5</span>, <span class='kw'>col</span> <span class='kw'>=</span> <span class='st'>"white"</span>, <span class='kw'>size</span> <span class='kw'>=</span> <span class='fl'>0.6</span>, <span class='kw'>width</span> <span class='kw'>=</span> <span class='fl'>0.7</span>)</div><img src='expressionsTCGA-9.png' alt='' width='540' height='400' /><div class='input'>


<span class='co'>## miRNASeq expressions </span>
<span class='fu'>library</span>(<span class='no'>RTCGA.miRNASeq</span>)
<span class='co'># miRNASeq has bcr_patienct_barcode in rownames...</span>
<span class='fu'>mutate</span>(<span class='no'>ACC.miRNASeq</span>,
   <span class='kw'>bcr_patient_barcode</span> <span class='kw'>=</span> <span class='fu'>substr</span>(<span class='fu'>rownames</span>(<span class='no'>ACC.miRNASeq</span>), <span class='fl'>1</span>, <span class='fl'>25</span>)) <span class='kw'>-&gt;</span> <span class='no'>ACC.miRNASeq.bcr</span>
<span class='fu'>mutate</span>(<span class='no'>CESC.miRNASeq</span>,
   <span class='kw'>bcr_patient_barcode</span> <span class='kw'>=</span> <span class='fu'>substr</span>(<span class='fu'>rownames</span>(<span class='no'>CESC.miRNASeq</span>), <span class='fl'>1</span>, <span class='fl'>25</span>)) <span class='kw'>-&gt;</span> <span class='no'>CESC.miRNASeq.bcr</span>
<span class='fu'>mutate</span>(<span class='no'>CHOL.miRNASeq</span>,
   <span class='kw'>bcr_patient_barcode</span> <span class='kw'>=</span> <span class='fu'>substr</span>(<span class='fu'>rownames</span>(<span class='no'>CHOL.miRNASeq</span>), <span class='fl'>1</span>, <span class='fl'>25</span>)) <span class='kw'>-&gt;</span> <span class='no'>CHOL.miRNASeq.bcr</span>
<span class='fu'>mutate</span>(<span class='no'>LAML.miRNASeq</span>,
   <span class='kw'>bcr_patient_barcode</span> <span class='kw'>=</span> <span class='fu'>substr</span>(<span class='fu'>rownames</span>(<span class='no'>LAML.miRNASeq</span>), <span class='fl'>1</span>, <span class='fl'>25</span>)) <span class='kw'>-&gt;</span> <span class='no'>LAML.miRNASeq.bcr</span>
<span class='fu'>mutate</span>(<span class='no'>PAAD.miRNASeq</span>,
   <span class='kw'>bcr_patient_barcode</span> <span class='kw'>=</span> <span class='fu'>substr</span>(<span class='fu'>rownames</span>(<span class='no'>PAAD.miRNASeq</span>), <span class='fl'>1</span>, <span class='fl'>25</span>)) <span class='kw'>-&gt;</span> <span class='no'>PAAD.miRNASeq.bcr</span>
<span class='fu'>mutate</span>(<span class='no'>THYM.miRNASeq</span>,
   <span class='kw'>bcr_patient_barcode</span> <span class='kw'>=</span> <span class='fu'>substr</span>(<span class='fu'>rownames</span>(<span class='no'>THYM.miRNASeq</span>), <span class='fl'>1</span>, <span class='fl'>25</span>)) <span class='kw'>-&gt;</span> <span class='no'>THYM.miRNASeq.bcr</span>
<span class='fu'>mutate</span>(<span class='no'>LGG.miRNASeq</span>,
   <span class='kw'>bcr_patient_barcode</span> <span class='kw'>=</span> <span class='fu'>substr</span>(<span class='fu'>rownames</span>(<span class='no'>LGG.miRNASeq</span>), <span class='fl'>1</span>, <span class='fl'>25</span>)) <span class='kw'>-&gt;</span> <span class='no'>LGG.miRNASeq.bcr</span>
<span class='fu'>mutate</span>(<span class='no'>STAD.miRNASeq</span>,
   <span class='kw'>bcr_patient_barcode</span> <span class='kw'>=</span> <span class='fu'>substr</span>(<span class='fu'>rownames</span>(<span class='no'>STAD.miRNASeq</span>), <span class='fl'>1</span>, <span class='fl'>25</span>)) <span class='kw'>-&gt;</span> <span class='no'>STAD.miRNASeq.bcr</span>


<span class='fu'>expressionsTCGA</span>(<span class='no'>ACC.miRNASeq.bcr</span>, <span class='no'>CESC.miRNASeq.bcr</span>, <span class='no'>CHOL.miRNASeq.bcr</span>,
             <span class='no'>LAML.miRNASeq.bcr</span>, <span class='no'>PAAD.miRNASeq.bcr</span>, <span class='no'>THYM.miRNASeq.bcr</span>,
             <span class='no'>LGG.miRNASeq.bcr</span>, <span class='no'>STAD.miRNASeq.bcr</span>,
  <span class='kw'>extract.cols</span> <span class='kw'>=</span> <span class='fu'>c</span>(<span class='st'>"machine"</span>, <span class='st'>"hsa-mir-101-1"</span>, <span class='st'>"miRNA_ID"</span>)) <span class='kw'>%&gt;%</span>
                <span class='fu'>rename</span>(<span class='kw'>cohort</span> <span class='kw'>=</span> <span class='no'>dataset</span>) <span class='kw'>%&gt;%</span>
   <span class='fu'>filter</span>(<span class='no'>miRNA_ID</span> <span class='kw'>==</span> <span class='st'>"read_count"</span>) <span class='kw'>%&gt;%</span>
   <span class='fu'>select</span>(-<span class='no'>bcr_patient_barcode</span>, -<span class='no'>miRNA_ID</span>) <span class='kw'>%&gt;%</span>
   <span class='fu'>gather</span>(<span class='kw'>key</span> <span class='kw'>=</span> <span class='st'>"key"</span>, <span class='kw'>value</span> <span class='kw'>=</span> <span class='st'>"value"</span>, -<span class='no'>cohort</span>, -<span class='no'>machine</span>) <span class='kw'>%&gt;%</span>
   <span class='fu'>mutate</span>(<span class='kw'>value</span> <span class='kw'>=</span> <span class='fu'>as.numeric</span>(<span class='no'>value</span>)) <span class='kw'>%&gt;%</span>
   <span class='fu'>ggplot</span>(<span class='fu'>aes</span>(<span class='kw'>x</span> <span class='kw'>=</span> <span class='no'>cohort</span>,
              <span class='kw'>y</span> <span class='kw'>=</span> <span class='fu'>log1p</span>(<span class='no'>value</span>),
              <span class='kw'>fill</span> <span class='kw'>=</span> <span class='fu'>as.factor</span>(<span class='no'>machine</span>)) )+
   <span class='fu'>geom_boxplot</span>() +
   <span class='fu'><a href='theme_RTCGA.html'>theme_RTCGA</a></span>(<span class='kw'>base_size</span> <span class='kw'>=</span> <span class='fl'>13</span>) +
   <span class='fu'>coord_flip</span>() +
   <span class='fu'>theme</span>(<span class='kw'>legend.position</span> <span class='kw'>=</span> <span class='st'>"top"</span>) +
   <span class='fu'>scale_fill_brewer</span>(<span class='kw'>palette</span> <span class='kw'>=</span> <span class='st'>"Paired"</span>) +
   <span class='fu'>ggtitle</span>(<span class='st'>"hsa-mir-101-1"</span>)</div><div class='output co'>#&gt; <span class='message'>Scale for &#39;fill&#39; is already present. Adding another scale for &#39;fill&#39;, which</span>
#&gt; <span class='message'>will replace the existing scale.</span></div><img src='expressionsTCGA-12.png' alt='' width='540' height='400' /><div class='input'>

</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#note">Note</a></li>

      <li><a href="#issues">Issues</a></li>

      <li><a href="#see-also">See also</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    
Marcin Kosinski, <a href='mailto:m.p.kosinski@gmail.com'>m.p.kosinski@gmail.com</a>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Marcin Kosinski.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  </body>
</html>
