<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Read TCGA data to the tidy Format — readTCGA • RTCGA</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">

<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">


<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script>
<script src="../pkgdown.js"></script>

<!-- mathjax -->
<script src='https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">RTCGA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../articles/index.html">Articles</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/RTCGA/RTCGA">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/kosinski_rblog">
    <span class="fa fa-twitter"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

      <div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Read TCGA data to the tidy Format</h1>
    </div>

    
    <p><code>readTCGA</code> function allows to read unzipped files: 
<ul>
<li> clinical data - <code>Merge_Clinical.Level_1</code> 
     </li>
<li> rnaseq data (genes&#39; expressions) - <code>rnaseqv2__illuminahiseq_rnaseqv2</code>
     </li>
<li> genes&#39; mutations data - <code>Mutation_Packager_Calls.Level</code>
     </li>
<li> Reverse phase protein array data (RPPA) - <code>protein_normalization__data.Level_3</code>
     </li>
<li> Merge transcriptome agilent data (mRNA) - 
     <code>Merge_transcriptome__agilentg4502a_07_3__unc_edu__Level_3__unc_lowess_normalization_gene_level__data.Level_3</code>
     </li>
<li> miRNASeq data - 
     <code>Merge_mirnaseq__illuminaga_mirnaseq__bcgsc_ca__Level_3__miR_gene_expression__data.Level_3</code> or 
     <code>&quot;Merge_mirnaseq__illuminahiseq_mirnaseq__bcgsc_ca__Level_3__miR_gene_expression__data.Level_3&quot;</code>
     </li>
<li> methylation data - 
     <code>Merge_methylation__humanmethylation27</code>
     </li>
<li> isoforms data - 
     <code>Merge_rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_isoforms_normalized__data.Level_3</code> 
     </li>
<li> CNV data - <code>segmented_scna_minus_germline_cnv_hg19</code>
     </li>
</ul></p>

    <p>from TCGA project. Those files can be easily downloded with <a href='downloadTCGA.html'>downloadTCGA</a> function. See examples.</p>
    

    <pre><span class='fu'>readTCGA</span>(<span class='no'>path</span>, <span class='no'>dataType</span>, <span class='no'>...</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a> Arguments</h2>
    <dl class="dl-horizontal">
      <dt>path</dt>
      <dd>See details and examples.</dd>
      <dt>dataType</dt>
      <dd>One of <code>&#39;clinical&#39;, &#39;rnaseq&#39;, &#39;mutations&#39;, &#39;RPPA&#39;, &#39;mRNA&#39;, &#39;miRNASeq&#39;, &#39;methylation&#39;, &#39;isoforms&#39;, &#39;CNV&#39;</code> depending on which type of data user is trying to read in the tidy format.</dd>
      <dt>...</dt>
      <dd>Further arguments passed to the as.data.frame.</dd>
    </dl>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>An output is a <code>data.frame</code> with <code>dataType</code> data.</p>
    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>All cohort names can be checked using: <code> sub( x = names( infoTCGA() ), &#39;-counts&#39;, &#39;&#39;)</code>.</p>
    <p>Parameter <code>path</code> specification: 
<ul>
<li> If <code>dataType = &#39;clinical&#39;</code> a path to a <code>cancerType.clin.merged.txt</code> file. 
</li>
<li> If <code>dataType = &#39;mutations&#39;</code> a path to the unzziped folder <code>Mutation_Packager_Calls.Level</code> containing <code>.maf</code> files.
</li>
<li> If <code>dataType = &#39;rnaseq&#39;</code> a path to the uzziped file <code>rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_genes_normalized__data.Level</code>.
</li>
<li> If <code>dataType = &#39;RPPA&#39;</code> a path to the unzipped file in folder <code>protein_normalization__data.Level_3</code>.
</li>
<li> If <code>dataType = &#39;mRNA&#39;</code> a path to the unzipped file <code>cancerType.transcriptome__agilentg4502a_07_3__unc_edu__Level_3__unc_lowess_normalization_gene_level__data.data.txt</code>.
</li>
<li> If <code>dataType = &#39;miRNASeq&#39;</code> a path to unzipped files <code>cancerType.mirnaseq__illuminahiseq_mirnaseq__bcgsc_ca__Level_3__miR_gene_expression__data.data.txt</code> or <code>cancerType.mirnaseq__illuminaga_mirnaseq__bcgsc_ca__Level_3__miR_gene_expression__data.data.txt</code>
</li>
<li> If <code>dataType = &#39;methylation&#39;</code> a path to unzipped files <code>cancerType.methylation__humanmethylation27__jhu_usc_edu__Level_3__within_bioassay_data_set_function__data.data.txt</code>.
</li>
<li> If <code>dataType = &#39;isoforms&#39;</code> a path to unzipped files <code>cancerType.rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_isoforms_normalized__data.data.txt</code>.
</li>
<li> If <code>dataType = &#39;CNV&#39;</code> a path to unzipped files <code>cancerType.Merge_snp__genome_wide_snp_6__broad_mit_edu__Level_3__segmented_scna_minus_germline_cnv_hg18__seg.Level_3.txt</code>.
</li>
</ul></p>
    
    <h2 class="hasAnchor" id="issues"><a class="anchor" href="#issues"></a>Issues</h2>

    <p></p>
    <p>If you have any problems, issues or think that something is missing or is not
clear please post an issue on 
<a href = 'https://github.com/RTCGA/RTCGA/issues'>https://github.com/RTCGA/RTCGA/issues</a>.</p>
    
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <p><span class="pkg">RTCGA</span> website <a href = 'http://rtcga.github.io/RTCGA/Download.html'>http://rtcga.github.io/RTCGA/Download.html</a>.</p>
    <p>Other RTCGA: <code><a href='RTCGA-package.html'>RTCGA-package</a></code>,
  <code><a href='boxplotTCGA.html'>boxplotTCGA</a></code>, <code><a href='checkTCGA.html'>checkTCGA</a></code>,
  <code><a href='convertTCGA.html'>convertTCGA</a></code>, <code><a href='createTCGA.html'>createTCGA</a></code>,
  <code><a href='datasetsTCGA.html'>datasetsTCGA</a></code>, <code><a href='downloadTCGA.html'>downloadTCGA</a></code>,
  <code><a href='expressionsTCGA.html'>expressionsTCGA</a></code>, <code><a href='heatmapTCGA.html'>heatmapTCGA</a></code>,
  <code><a href='infoTCGA.html'>infoTCGA</a></code>, <code><a href='installTCGA.html'>installTCGA</a></code>,
  <code><a href='kmTCGA.html'>kmTCGA</a></code>, <code><a href='mutationsTCGA.html'>mutationsTCGA</a></code>,
  <code><a href='pcaTCGA.html'>pcaTCGA</a></code>, <code><a href='survivalTCGA.html'>survivalTCGA</a></code>,
  <code><a href='theme_RTCGA.html'>theme_RTCGA</a></code></p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='co'>## Not run: ------------------------------------</span>
<span class='co'>#  </span>
<span class='co'># </span>
<span class='co'># ##############</span>
<span class='co'># ##### clinical</span>
<span class='co'># ##############</span>
<span class='co'># </span>
<span class='co'># dir.create('data')</span>
<span class='co'># </span>
<span class='co'># # downloading clinical data</span>
<span class='co'># # dataset = "clinical" is default parameter so we may omit it</span>
<span class='co'># downloadTCGA(cancerTypes = c('BRCA', 'OV'),</span>
<span class='co'>#              destDir = 'data' )</span>
<span class='co'># # shorten paths so that they are shorter than 256 signs - windows issue</span>
<span class='co'>#  list.files("data", full.names = TRUE) %&gt;%</span>
<span class='co'>#    file.rename(to = substr(., start = 1, stop = 50))</span>
<span class='co'>#     </span>
<span class='co'># # reading datasets    </span>
<span class='co'># sapply(c('BRCA', 'OV'), function(element){</span>
<span class='co'>#  path &lt;- list.files('data', recursive = TRUE,</span>
<span class='co'>#                     full.names = TRUE, </span>
<span class='co'>#                     patten = "clin.merged.txt")</span>
<span class='co'>#  assign(value = readTCGA( path, 'clinical' ), </span>
<span class='co'>#         x = paste0(element, '.clin.data'),</span>
<span class='co'>#         envir = .GlobalEnv)})</span>
<span class='co'>#      </span>
<span class='co'># ############</span>
<span class='co'># ##### rnaseq</span>
<span class='co'># ############</span>
<span class='co'># </span>
<span class='co'># dir.create('data2')</span>
<span class='co'># </span>
<span class='co'># # downloading rnaseq data</span>
<span class='co'># downloadTCGA(cancerTypes = 'BRCA', </span>
<span class='co'>#              dataSet = 'Level_3__RSEM_genes_normalized',</span>
<span class='co'>#              destDir = 'data2')</span>
<span class='co'># </span>
<span class='co'># # shorten paths so that they are shorter than 256 signs - windows issue</span>
<span class='co'># list.files("data2", full.names = TRUE) %&gt;%</span>
<span class='co'>#    file.rename(to = substr(., start = 1, stop = 50))</span>
<span class='co'># </span>
<span class='co'># path_rnaseq &lt;- list.files('data2', recursive = TRUE,</span>
<span class='co'>#                           full.names = TRUE, </span>
<span class='co'>#                           patten = 'illuminahiseq')</span>
<span class='co'># readTCGA(path = pathRNA, dataType = 'rnaseq') -&gt; rnaseq_data</span>
<span class='co'># </span>
<span class='co'># </span>
<span class='co'># ###############</span>
<span class='co'># ##### mutations</span>
<span class='co'># ###############</span>
<span class='co'># </span>
<span class='co'># # Example directory in which untarred data will be stored</span>
<span class='co'># dir.create('data3')</span>
<span class='co'># </span>
<span class='co'># </span>
<span class='co'># downloadTCGA(cancerTypes = 'OV', </span>
<span class='co'>#              dataSet = 'Mutation_Packager_Calls.Level',</span>
<span class='co'>#              destDir = 'data3')</span>
<span class='co'># </span>
<span class='co'># # reading data</span>
<span class='co'># list.files('data3', recursive = TRUE) -&gt; directory</span>
<span class='co'># </span>
<span class='co'># readTCGA(directory, 'mutations') -&gt; mut_file</span>
<span class='co'># </span>
<span class='co'># #################</span>
<span class='co'># ##### methylation</span>
<span class='co'># #################</span>
<span class='co'># </span>
<span class='co'># # Example directory in which untarred data will be stored</span>
<span class='co'># dir.create('data4')</span>
<span class='co'># </span>
<span class='co'># # Download KIRP methylation data and store it in data4 folder</span>
<span class='co'># cancerType = "KIRP"</span>
<span class='co'># downloadTCGA(cancerTypes = cancerType,</span>
<span class='co'>#              dataSet = "Merge_methylation__humanmethylation27",</span>
<span class='co'>#              destDir = "data4")</span>
<span class='co'># </span>
<span class='co'># # Shorten path of subdirectory with KIRP methylation data</span>
<span class='co'># list.files(path = "data4", full.names = TRUE) %&gt;%</span>
<span class='co'>#     file.rename(to = file.path("data4", paste0(cancerType, ".methylation")))</span>
<span class='co'># </span>
<span class='co'># # Remove manifest.txt file</span>
<span class='co'># list.files(path = "data4", full.names = TRUE, </span>
<span class='co'>#            recursive = TRUE, pattern = "MANIFEST") %&gt;%</span>
<span class='co'>#            file.remove()</span>
<span class='co'># </span>
<span class='co'># # Read KIRP methylation data</span>
<span class='co'># path &lt;- list.files(path = "data4", full.names = TRUE, recursive = TRUE)</span>
<span class='co'># KIRP.methylation &lt;- readTCGA(path, dataType = "methylation")</span>
<span class='co'># </span>
<span class='co'># </span>
<span class='co'># ##########</span>
<span class='co'># ##### RPPA</span>
<span class='co'># ##########</span>
<span class='co'># </span>
<span class='co'># # Directory in which untarred data will be stored</span>
<span class='co'># dir.create('data5')</span>
<span class='co'># </span>
<span class='co'># # Download BRCA RPPA data and store it in data5 folder</span>
<span class='co'># cancerType = "BRCA"</span>
<span class='co'># downloadTCGA(cancerTypes = cancerType,</span>
<span class='co'>#              dataSet = "protein_normalization__data.Level_3",</span>
<span class='co'>#              destDir = "data5")</span>
<span class='co'># </span>
<span class='co'># # Shorten path of subdirectory with BRCA RPPA data</span>
<span class='co'># list.files(path = "data5", full.names = TRUE) %&gt;%</span>
<span class='co'>#     file.rename(from = ., to = file.path("data5", paste0(cancerType, ".RPPA")))</span>
<span class='co'># </span>
<span class='co'># # Remove manifest.txt file</span>
<span class='co'># list.files(path = "data5", full.names = TRUE,</span>
<span class='co'>#            recursive = TRUE, pattern = "MANIFEST") %&gt;%</span>
<span class='co'>#            file.remove()</span>
<span class='co'># </span>
<span class='co'># # Read BRCA RPPA data</span>
<span class='co'># path &lt;- list.files(path = "data5", full.names = TRUE, recursive = TRUE) </span>
<span class='co'># BRCA.RPPA &lt;- readTCGA(path, dataType = "RPPA")</span>
<span class='co'># </span>
<span class='co'># </span>
<span class='co'># ##########</span>
<span class='co'># ##### mRNA</span>
<span class='co'># ##########</span>
<span class='co'># </span>
<span class='co'># # Directory in which untarred data will be stored</span>
<span class='co'># dir.create('data6')</span>
<span class='co'># </span>
<span class='co'># # Download UCEC mRNA data and store it in data6 folder</span>
<span class='co'># cancerType = "UCEC"</span>
<span class='co'># downloadTCGA(cancerTypes = cancerType,</span>
<span class='co'>#              dataSet = "agilentg4502a_07_3__unc_edu__Level_3",</span>
<span class='co'>#              destDir = "data6")</span>
<span class='co'># </span>
<span class='co'># # Shorten path of subdirectory with UCEC mRNA data</span>
<span class='co'># list.files(path = "data6", full.names = TRUE) %&gt;%</span>
<span class='co'>#     file.rename(from = ., to = file.path("data6",paste0(cancerType, ".mRNA")))</span>
<span class='co'># </span>
<span class='co'># # Remove manifest.txt file</span>
<span class='co'># list.files(path = "data6", full.names = TRUE,</span>
<span class='co'>#            recursive = TRUE, pattern = "MANIFEST") %&gt;%</span>
<span class='co'>#            file.remove()</span>
<span class='co'># </span>
<span class='co'># # Read UCEC mRNA data</span>
<span class='co'># path &lt;- list.files(path = "data6", full.names = TRUE, recursive = TRUE) </span>
<span class='co'># UCEC.mRNA &lt;- readTCGA(path, dataType = "mRNA")</span>
<span class='co'># </span>
<span class='co'># ##############</span>
<span class='co'># ##### miRNASeq</span>
<span class='co'># ##############</span>
<span class='co'># </span>
<span class='co'># # Directory in which untarred data will be stored</span>
<span class='co'># dir.create('data7')</span>
<span class='co'># </span>
<span class='co'># # Download BRCA miRNASeq data and store it in data7 folder</span>
<span class='co'># # Remember that miRNASeq data are produced by two machines:</span>
<span class='co'># # Illumina Genome Analyzer and Illumina HiSeq 2000 machines</span>
<span class='co'># cancerType &lt;- "BRCA"</span>
<span class='co'># downloadTCGA(cancerTypes = cancerType,</span>
<span class='co'># dataSet = paste0("Merge_mirnaseq__illuminaga_mirnaseq__bcgsc",</span>
<span class='co'>#                 "_ca__Level_3__miR_gene_expression__data.Level_3"),</span>
<span class='co'>#              destDir = "data7")</span>
<span class='co'># </span>
<span class='co'># downloadTCGA(cancerTypes = cancerType,</span>
<span class='co'># dataSet = paste0("Merge_mirnaseq__illuminahiseq_mirnaseq__",</span>
<span class='co'>#                  "bcgsc_ca__Level_3__miR_gene_expression__data.Level_3"),</span>
<span class='co'>#              destDir = "data7")</span>
<span class='co'># </span>
<span class='co'># # Shorten path of subdirectory with BRCA miRNASeq data</span>
<span class='co'># list.files(path = "data7", full.names = TRUE) %&gt;%</span>
<span class='co'>#     sapply(function(path){</span>
<span class='co'>#         if (grepl(pattern = "illuminaga", path)){</span>
<span class='co'>#             file.rename(from = grep(pattern = "illuminaga", path, value = TRUE),</span>
<span class='co'>#                         to = file.path("data7",paste0(cancerType, ".miRNASeq.illuminaga")))</span>
<span class='co'>#         } else if (grepl(pattern = "illuminahiseq", path)){</span>
<span class='co'>#             file.rename(from = grep(pattern = "illuminahiseq", path, value = TRUE),</span>
<span class='co'>#                         to = file.path("data7",paste0(cancerType, ".miRNASeq.illuminahiseq")))</span>
<span class='co'>#         }</span>
<span class='co'>#     })</span>
<span class='co'>#     </span>
<span class='co'># # Remove manifest.txt file</span>
<span class='co'># list.files(path = "data6", full.names = TRUE,</span>
<span class='co'>#            recursive = TRUE, pattern = "MANIFEST") %&gt;%</span>
<span class='co'>#            file.remove()</span>
<span class='co'># </span>
<span class='co'># # Read BRCA miRNASeq data</span>
<span class='co'># path &lt;- list.files(path = "data7", full.names = TRUE, recursive = TRUE)</span>
<span class='co'># path_illuminaga &lt;- grep("illuminaga", path, fixed = TRUE, value = TRUE)</span>
<span class='co'># path_illuminahiseq &lt;- grep("illuminahiseq", path, fixed = TRUE, value = TRUE)</span>
<span class='co'># </span>
<span class='co'># BRCA.miRNASeq.illuminaga &lt;- readTCGA(path_illuminaga, dataType = "miRNASeq")</span>
<span class='co'># BRCA.miRNASeq.illuminahiseq &lt;- readTCGA(path_illuminahiseq, dataType = "miRNASeq")</span>
<span class='co'># </span>
<span class='co'># BRCA.miRNASeq.illuminaga &lt;- cbind(machine = "Illumina Genome Analyzer",</span>
<span class='co'>#                                   BRCA.miRNASeq.illuminaga)</span>
<span class='co'># BRCA.miRNASeq.illuminahiseq &lt;- cbind(machine = "Illumina HiSeq 2000",</span>
<span class='co'>#                                      BRCA.miRNASeq.illuminahiseq)</span>
<span class='co'># </span>
<span class='co'># BRCA.miRNASeq &lt;- rbind(BRCA.miRNASeq.illuminaga, BRCA.miRNASeq.illuminahiseq)</span>
<span class='co'># </span>
<span class='co'># ##############</span>
<span class='co'># ##### isoforms</span>
<span class='co'># ##############</span>
<span class='co'># </span>
<span class='co'># # Directory in which untarred data will be stored</span>
<span class='co'># dir.create('data8')</span>
<span class='co'># </span>
<span class='co'># # Download ACC isoforms data and store it in data8 folder</span>
<span class='co'># cancerType = "ACC"</span>
<span class='co'># downloadTCGA(cancerTypes = cancerType,</span>
<span class='co'># dataSet = paste0("Merge_rnaseqv2__illuminahiseq_rnaseqv2__unc",</span>
<span class='co'>#                  "_edu__Level_3__RSEM_isoforms_normalized__data.Level_3"),</span>
<span class='co'>#              destDir = "data8")</span>
<span class='co'># </span>
<span class='co'># # Shorten path of subdirectory with ACC isoforms data</span>
<span class='co'># list.files(path = "data8", full.names = TRUE) %&gt;%</span>
<span class='co'>#     file.rename(from = ., to = file.path("data8",paste0(cancerType, ".isoforms")))</span>
<span class='co'># </span>
<span class='co'># # Remove manifest.txt file</span>
<span class='co'># list.files(path = "data6", full.names = TRUE,</span>
<span class='co'>#            recursive = TRUE, pattern = "MANIFEST") %&gt;%</span>
<span class='co'>#            file.remove()</span>
<span class='co'># </span>
<span class='co'># # Read ACC isoforms data</span>
<span class='co'># path &lt;- list.files(path = "data8", full.names = TRUE, recursive = TRUE) </span>
<span class='co'># ACC.isoforms &lt;- readTCGA(path, dataType = "isoforms")</span>
<span class='co'># </span>
<span class='co'>## ---------------------------------------------</span></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#details">Details</a></li>

      <li><a href="#issues">Issues</a></li>

      <li><a href="#see-also">See also</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    
Marcin Kosinski, <a href='mailto:m.p.kosinski@gmail.com'>m.p.kosinski@gmail.com</a>

Witold Chodor, <a href='mailto:witoldchodor@gmail.com'>witoldchodor@gmail.com</a>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Marcin Kosinski.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  </body>
</html>
